packages/rum-react/src/domain/reactRouterV6/useRoutes.ts

```diff
@@ -0,0 +1,19 @@
+import { useRef } from 'react'
+import { matchRoutes, useLocation, useRoutes as originalUseRoutes } from 'react-router-dom'
+import { startReactRouterView } from './startReactRouterView'
+
+export const useRoutes: typeof originalUseRoutes = (routes, locationArg) => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏This implementation does not allow to have multiple instrumentation of `useRoute`. For instance if Sentry had the same approach the customer could not use both Sentry and Datadog (weird scenario but it is an example :). We could use a similar approach as Sentry using a HOF:
```ts
const useSentryRoutes = Sentry.wrapUseRoutes(useRoutes);
const useDatadogRoutes = Datadog.wrapUseRoutes(useSentryRoutes);
```

Reply:
This is possible no?

```ts
import { useRoutes} from '@datadog/browser-rum-react/react-router-6'

const useSentryRoutes = Sentry.wrapUseRoutes(useRoutes)
```

Reply:
Indeed, this might be an issue if the customer uses another library that also provides a patched `useRoute` instead of providing a wrapping utility. I think this might be an edge case, and if a customer ever face that issue, we can always provide a wrapping utility (beside the already-wrapped `useRoutes`).

Reply:
Okay works for me 🙂